### PR TITLE
docs: update example publishing scripts

### PR DESCRIPTION
**Pull request by bethesque**
Show best practice for consumer versions and tags.

### CODE REVIEW

**Comment by bethesque** [comment]
https://github.com/pact-foundation/pact-js/issues/442

File: examples/serverless/publish.js [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
+
+//Usually, you would just use the CI env vars, but to allow these examples to run from
+//local development machines, we'll fall back to the git command when the env vars aren't set.
+const gitSha = process.env.TRAVIS_COMMIT || exec("git rev-parse HEAD")
+const branch =
+  process.env.TRAVIS_BRANCH || exec("git rev-parse --abbrev-ref HEAD")
+
 const opts = {
-  pactFilesOrDirs: [path.resolve(__dirname, "pacts/")],
+  pactFilesOrDirs: ["./pacts/"],
```
Comment (TimothyJones) [inline examples/serverless/publish.js:22]: Unless there's a subtlety I am missing, I think the `__dirname` version was more robust, as it is defined to be the same place as this script.

Comment (mefellows) [reply examples/serverless/publish.js:22]: That's correct Tim. If somebody runs the script from a different directory it will fail.

**Comment by mefellows** [comment]
Thanks Beth. I've updated to make use of the new Publisher (which now completely hides pact node) and reverted to the uglier, but more resilient, `path.resolve(...)`. People do funny things, like run it from node directly, run npm commands from random directories - it's just easier for us this way!

**Pull request merged**